### PR TITLE
Disableable authentication listener

### LINKED ISSUE #1658: There should be a way to disable AuthenticationListener
I am on v2-alpha, and there is no simple way to disable automatic authentication.
Looking for one of the ways to disable it:
- via fos_user configuration
- via class override

Any ideas?

### PR DESCRIPTION
**Pull request by geoffrey-brier**
Fixes #1658

### CODE REVIEW

File: DependencyInjection/FOSUserExtension.php [hunk 1]
```diff
@@ -67,10 +67,14 @@ public function load(array $configs, ContainerBuilder $container)
             $definition->setFactory(array(new Reference('fos_user.doctrine_registry'), 'getManager'));
         }
 
-        foreach (array('validator', 'security', 'util', 'mailer', 'listeners') as $basename) {
+        foreach (array('validator', 'security', 'util', 'mailer') as $basename) {
             $loader->load(sprintf('%s.xml', $basename));
         }
 
+        if ($config['use_authentication_listener']) {
+            $loader->load('listeners.xml');
```
Comment (core23) [inline DependencyInjection/FOSUserExtension.php]: This would disable all listeners

Comment (stof) [inline DependencyInjection/FOSUserExtension.php]: this is indeed wrong. You should disable only the relevant listener (by removing the service)

**Comment by stof** [comment]
the doc must be updated too

**Comment by XWB** [comment]
@geoffrey-brier Can you make the requested changes?

**Comment by geoffrey-brier** [comment]
@XWB My bad, completely forgot about this one. Is this ok ?

**Comment by XWB** [comment]
Will update the docs later.

File: DependencyInjection/FOSUserExtension.php [hunk 2]
```diff
@@ -71,6 +71,10 @@ public function load(array $configs, ContainerBuilder $container)
             $loader->load(sprintf('%s.xml', $basename));
         }
 
+        if ($config['use_authentication_listener']) {
+            $container->removeDefinition('fos_user.listener.authentication');
```
Comment (StudioMaX) [inline DependencyInjection/FOSUserExtension.php]: Listener will be disabled when the "use_authentication_listener" is TRUE? Is that correct?

Comment (geoffrey-brier) [inline DependencyInjection/FOSUserExtension.php]: @StudioMaX good catch ...

**Pull request merged**